zipline/data/data_portal.py

```diff
@@ -1441,17 +1365,11 @@ def _apply_all_adjustments(self, data, asset, dts, field,
                 True
             )
 
-            data *= price_adj_factor
-
-            # if anything is zero, it's a missing bar, so replace it with NaN.
-            # we only want to do this for non-volume fields, because a missing
-            # volume should be 0.
-            data[data == 0] = np.NaN
-
-        np.around(data, 3, out=data)
+            if price_adj_factor is not None:
```

Comment:
this is likely going to conflict with lazy-mainline, where this method has been renamed and tweaked a bit.